river/Cursor.zig

```diff
@@ -1049,21 +1050,33 @@ pub fn checkFocusFollowsCursor(self: *Self) void {
     if (self.seat.drag == .pointer) return;
     if (server.config.focus_follows_cursor == .disabled) return;
     if (self.surfaceAt()) |result| {
-        if (server.config.focus_follows_cursor == .always or
-            self.seat.wlr_seat.pointer_state.focused_surface != result.surface)
-        {
-            switch (result.parent) {
-                .view => |view| {
-                    if (self.seat.focused != .view or self.seat.focused.view != view) {
-                        self.seat.focusOutput(view.output);
-                        self.seat.focus(view);
-                        server.root.startTransaction();
-                    }
-                },
-                .layer_surface, .lock_surface => {},
-                .xwayland_override_redirect => assert(build_options.xwayland),
-            }
+        switch (result.parent) {
+            .view => |view| {
+                // Don't re-focus the last focused view when the mode is .normal
+                if (server.config.focus_follows_cursor == .normal and
+                    view == self.last_followed_view) return;
```

Comment:
Could you explain why we need this `last_followed_view` field instead of the current `self.seat.wlr_seat.pointer_state.focused_surface != result.surface` check?

Reply:
because the focused_surface is set when the cursor is over the input_region, which we don't want